docs/sql-programming-guide.md

```diff
@@ -1898,6 +1898,7 @@ working with timestamps in `pandas_udf`s to get the best performance, see
   - Since Spark 2.4, File listing for compute statistics is done in parallel by default. This can be disabled by setting `spark.sql.parallelFileListingInStatsComputation.enabled` to `False`.
   - Since Spark 2.4, Metadata files (e.g. Parquet summary files) and temporary files are not counted as data files when calculating table size during Statistics computation.
   - Since Spark 2.4, empty strings are saved as quoted empty strings `""`. In version 2.3 and earlier, empty strings are equal to `null` values and do not reflect to any characters in saved CSV files. For example, the row of `"a", null, "", 1` was writted as `a,,,1`. Since Spark 2.4, the same row is saved as `a,,"",1`. To restore the previous behavior, set the CSV option `emptyValue` to empty (not quoted) string.  
+  - Since Spark 2.4 load command from local filesystem supports wildcards in the folder level paths(e.g. LOAD DATA LOCAL INPATH 'tmp/folder*/). Now onwards normal space convention can be used in folder/file names (e.g. LOAD DATA INPATH 'tmp/folderName/file Name.csv), Older versions space in folder/file names has been represented using '%20'(e.g. LOAD DATA INPATH 'tmp/folderName/myFile%20Name.csv), this usage will not be supported from spark 2.4 version.
```

Comment:
Is it specific to the local file system?
Can this text add a quick example of using `?` too? this wildcard syntax is not regex syntax.
From your previous analysis, spaces in paths didn't work before even if escaped with `%20`. Shouldn't we just say that, additionally, special characters in paths like spaces should work now, and give the example?

Reply:
<<From your previous analysis, spaces in paths didn't work before even if escaped with %20. Shouldn't we just say that>>
@srowen  I modified the statement please recheck once and let me know for any suggestions.

Reply:
Is it specific to the local file system? << Yes, , its specific to local file system as in hdfs user can provide wildcard character in folder level also, for local file system folder level support was not there and error will be thrown)
Can this text add a quick example of using ? too?<< Yes i added the same>>

Reply:
I think this text doesn't describe the change then. The new functionality is that wildcards work at all levels in both local and remote file systems, right? This also says that `%20` escaping used to work, but your results show it didn't. That doesn't seem like a change in behavior. This text also has typos and spacing problems. To be clear, here is the text I suggest:

Since Spark 2.4, the `LOAD DATA` command supports wildcard characters `?` and `*`, which match any one character, and zero or more characters, respectively. Example: `LOAD DATA INPATH '/tmp/folder*/` or `LOAD DATA INPATH /tmp/part-?`. Characters like spaces also now work in paths: `LOAD DATA INPATH `/tmp/folder name/`.